pkg/distributors/test/MultiDistributor.test.ts

```diff
@@ -32,7 +34,16 @@ describe('MultiDistributor', () => {
   });
 
   sharedBeforeEach('deploy distributor', async () => {
-    distributor = await MultiDistributor.create();
+    vault = await Vault.create();
+    distributor = await MultiDistributor.create(vault);
+
+    // Authorise distributor to use users' vault token approvals
+    const action = await actionId(vault.instance, 'manageUserBalance');
+    await vault.grantRoleGlobally(action, distributor);
+    await vault.setRelayerApproval(user1, distributor, true);
+    await vault.setRelayerApproval(user2, distributor, true);
+    await vault.setRelayerApproval(user3, distributor, true);
+    await vault.setRelayerApproval(other, distributor, true);
```

Comment:
Coupled with what I mentioned above, I am now unsure about this feature. Maybe #1009 is not a great idea after all? There, I mention something about relayers, but that comment doesn't quite make sense and just seems incorrect.

Reply:
#1009 makes sense to me? BPT will always have an existing approval on the vault but in order to use this we need the staking contract to be a relayer so it can make use of this.

Users are then required to approve the staking contract as a relayer but this is a one-off action which will avoid them having to do a similar approval for each token.

This seems sound to me. But agreed, we'll have to open a backdoor for token contracts to autostake.